src/commands/build.ts

```diff
@@ -85,36 +83,58 @@ export class BuildCommand implements Command {
     },
   ];
 
-  run(options: CommandOptions, config: ProjectConfig): Promise<any> {
+  async run(options: CommandOptions, config: ProjectConfig): Promise<any> {
     // Defer dependency loading until this specific command is run
-    const build = require('../build/build').build;
+    let build = require('../build/build').build;
+    const hasCLIArgumentsPassed = this.args.some(arg => {
+      return typeof options[arg.name] !== 'undefined';
+    });
 
-    let buildOptions: BuildOptions = {
-      addServiceWorker: options['add-service-worker'],
-      swPrecacheConfig: options['sw-precache-config'],
-      insertPrefetchLinks: options['insert-prefetch-links'],
-      bundle: options['bundle'],
-      html: {
-        minify: !!options['html-minify'],
-      },
-      css: {
-        minify: !!options['css-minify'],
-      },
-      js: {
-        minify: !!options['js-minify'],
-        compile: !!options['js-compile'],
-      },
-    };
+    // Support passing a custom build function via options.env
+    if (options['env'] && options['env'].build) {
+      logger.debug('build function passed in options, using that for build');
+      build = options['env'].build;
+    }
 
-    logger.debug('building with options', buildOptions);
+    const mainBuildDirectory = 'build/';
+    logger.info(`Clearing ${mainBuildDirectory} directory...`);
+    await del([mainBuildDirectory]);
 
-    if (options['env'] && options['env'].build) {
-      logger.debug('env.build() found in options');
-      logger.debug('building via env.build()...');
-      return options['env'].build(buildOptions, config);
+    // If any the build command flags were passed as CLI arguments, generate
+    // a single build based on those flags alone.
+    if (hasCLIArgumentsPassed) {
+      return build(
+          {
+            addServiceWorker: options['add-service-worker'],
+            swPrecacheConfig: options['sw-precache-config'],
+            insertPrefetchLinks: options['insert-prefetch-links'],
+            bundle: options['bundle'],
+            html: {
+              minify: !!options['html-minify'],
+            },
+            css: {
+              minify: !!options['css-minify'],
+            },
+            js: {
+              minify: !!options['js-minify'],
+              compile: !!options['js-compile'],
+            },
+          },
+          config);
+    }
+
+    // If any builds were defined or configured via the project config,
+    // generate a build for each configuration.
+    if (config.builds) {
+      return Promise.all(
+          config.builds.map((buildOptions: ProjectBuildOptions, i: number) => {
+            buildOptions.name = buildOptions.name || `build-${i + 1}`;
```

Comment:
Fixed in polymer-project-config https://github.com/Polymer/polymer-project-config/pull/5